packages/dashboard/src/contexts/DashContext/state.ts

```diff
@@ -0,0 +1,122 @@
+import type { providers } from "ethers";
+import type {
+  DashboardMessageBusClient,
+  ReceivedMessageLifecycle
+} from "@truffle/dashboard-message-bus-client";
+import {
+  isDashboardProviderMessage,
+  isInvalidateMessage,
+  isLogMessage,
+  isDebugMessage
+} from "@truffle/dashboard-message-bus-common";
+import type {
+  Message,
+  DashboardProviderMessage
+} from "@truffle/dashboard-message-bus-common";
+import {
+  messageNeedsInteraction,
+  messageIsUnsupported,
+  rejectMessage,
+  confirmMessage
+} from "src/utils/dash";
+import type noticeContentType from "src/components/composed/Notice/noticeContentType";
+
+type actionType =
```

Comment:
Types should be capitalized, since they're akin to classes, not instances

Reply:
I'd also say that you could do something like this:

```typescript
interface Action {
  type: string;
}
```

and just do:

```typescript
interface SetClientAction extends Action {
  type: "set-client";
  data: DashboardMessageBusClient;
}
```

(`extends Action` is maybe unnecessary here, but it might be helpful cause then it will typecheck)

Reply:
Yup I've [capitalized types and interfaces](https://github.com/trufflesuite/truffle/commit/486efcb9cae84609c6966dc64a8a9963384a17a3)! I did end up having [a `BaseAction` that others extend off of](https://github.com/trufflesuite/truffle/blob/dash-woah/packages/dashboard/src/contexts/DashContext/types/Action.ts) to imply that `AnAction` and `AnotherAction` are "similar", and not unrelated.